fdbserver/TagPartitionedLogSystem.actor.cpp

```diff
@@ -20,17 +20,41 @@
 
 #include "fdbserver/TagPartitionedLogSystem.actor.h"
 
+#include <utility>
+
 #include "flow/actorcompiler.h" // This must be the last #include.
 
-ACTOR Future<Version> minVersionWhenReady(Future<Void> f, std::vector<Future<TLogCommitReply>> replies) {
-	wait(f);
-	Version minVersion = std::numeric_limits<Version>::max();
-	for (auto& reply : replies) {
-		if (reply.isReady() && !reply.isError()) {
-			minVersion = std::min(minVersion, reply.get().version);
+ACTOR Future<Version> minVersionWhenReady(Future<Void> f,
+                                          std::vector<std::pair<UID, Future<TLogCommitReply>>> replies) {
+	try {
+		wait(f);
+		Version minVersion = std::numeric_limits<Version>::max();
+		for (const auto& [_tlogID, reply] : replies) {
+			if (reply.isReady() && !reply.isError()) {
+				minVersion = std::min(minVersion, reply.get().version);
+			}
+		}
+		return minVersion;
+	} catch (Error& err) {
+		if (err.code() == error_code_operation_cancelled) {
+			TraceEvent(SevInfo, "TLogPushCancelled");
+			int index = 0;
+			for (const auto& [tlogID, reply] : replies) {
+				if (reply.isReady()) {
+					continue;
```

Comment:
add `index++` before "continue"

Reply:
I think this is unnecessary since the order of tLogID is not quite useful.

Reply:
Agree. I have no strong opinion, but the code give a false sense of using the correct `index`.